src/librustdoc/html/render/print_item.rs

```diff
@@ -1102,10 +1102,16 @@ fn item_trait(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     );
 }
 
-fn item_trait_alias(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean::TraitAlias) {
-    wrap_item(w, |w| {
+fn item_trait_alias(
+    w: &mut impl fmt::Write,
+    cx: &mut Context<'_>,
+    it: &clean::Item,
+    t: &clean::TraitAlias,
+) {
+    let mut buffer = Buffer::new();
+    wrap_item(&mut buffer, |buffer| {
```

Comment:
Something I just thought about: is it actually necessary to pass `Buffer` as argument to `wrap_item`? If we're forced to, would it make sense to have another `wrap_item_write` which would take `impl fmt::Write` instead?

Reply:
Yeah we could actually write another `wrap_item_write`  that would take in impl fmt::Write and it would actually reduce the overhead of creating a buffer every single time.
```
fn wrap_item_write<F>(w: &mut impl fmt::Write, f: F)
where
    F: FnOnce(&mut impl fmt::Write),
{
    write!(w, r#"<pre class="rust item-decl"><code>"#).unwrap();
    f(w);
    write!(w, "</code></pre>").unwrap();
}
```

Reply:
Then please do.

Reply:
Lets do it in this PR itself and then we can merge the rest, or do you want me to create a new PR to implement that function

Reply:
You can do it in this PR directly.